dynamo/tools/clustering.py

```diff
@@ -494,8 +505,12 @@ def cluster_community_from_graph(graph=None, graph_sparse_matrix=None, method="l
         if "initial_membership" in kwargs:
             logger.info("Detecting community with initial_membership input from caller")
             initial_membership = kwargs["initial_membership"]
-        if "weights" in kwargs:
-            weights = kwargs["weights"]
+            kwargs.pop("initial_membership")
```

Comment:
why do you pop the parameter initial_membership here? if pop from a dictionary, the key/value will be gone